samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java

```diff
@@ -186,14 +160,30 @@ private void startSubscription() {
    */
   void startConsumer() {
     // set the offset for each TopicPartition
-    if (topicPartitionToStartpointMap.size() <= 0) {
+    if (topicPartitionsToOffset.size() <= 0) {
       LOG.error ("{}: Consumer is not subscribed to any SSPs", this);
     }
 
-    topicPartitionToStartpointMap.forEach((topicPartition, startpoint) -> {
-      Partition partition = new Partition(topicPartition.partition());
-      SystemStreamPartition systemStreamPartition = new SystemStreamPartition(systemName, topicPartition.topic(), partition);
-      startpoint.apply(systemStreamPartition, kafkaStartpointRegistrationHandler);
+    topicPartitionsToOffset.forEach((tp, startingOffsetString) -> {
+      long startingOffset = Long.valueOf(startingOffsetString);
+
+      try {
+        synchronized (kafkaConsumer) {
+          kafkaConsumer.seek(tp, startingOffset); // this value should already be the 'upcoming' value
```

Comment:
Comment doesn't seem to match functionality. The `startingOffset` might not be the "upcoming" offset.